pennylane/templates/subroutines/qubitization.py

```diff
@@ -0,0 +1,147 @@
+
+    **Example**
+
+    This operator, when applied in conjunction with QPE, allows computing the eigenvalue of an eigenvector of the Hamiltonian.
+
+    .. code-block::
+
+        H = qml.dot([0.1, 0.3, -0.3], [qml.Z(0), qml.Z(1), qml.Z(0) @ qml.Z(2)])
+
+        @qml.qnode(qml.device("default.qubit"))
+        def circuit():
+
+          # initiate the eigenvector
+          qml.PauliX(2)
+
+          # apply QPE (used iterative qpe here)
+          measurements = qml.iterative_qpe(
+                         qml.Qubitization(H, control = [3,4]), ancilla = 5, iters = 3
+                         )
+          return qml.probs(op = measurements)
+
+        output = circuit()
+
+        # post-processing
+        lamb = sum([abs(c) for c in H.terms()[0]])
+
+    .. code-block:: pycon
+
+        >>> print("eigenvalue: ", lamb * np.cos(2 * np.pi * (np.argmax(output)) / 8))
+        eigenvalue: 0.7
+    """
+
+    def __init__(self, hamiltonian, control, id=None):
+        wires = hamiltonian.wires + qml.wires.Wires(control)
+
+        self._hyperparameters = {
+            "hamiltonian": hamiltonian,
+            "control": control,
+        }
+
+        super().__init__(wires=wires, id=id)
+
+    @staticmethod
+    def compute_decomposition(*_, **kwargs):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators (static method).
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+        .. seealso:: :meth:`~.Qubitization.decomposition`.
+
+        Args:
+            *params (list): trainable parameters of the operator, as stored in the ``parameters`` attribute
+            wires (Iterable[Any], Wires): wires that the operator acts on
+            **hyperparams (dict): non-trainable hyperparameters of the operator, as stored in the ``hyperparameters`` attribute
+
+        Returns:
+            list[Operator]: decomposition of the operator
+
+        **Example:**
+
+        >>> print(qml.Qubitization.compute_decomposition(hamiltonian = 0.1 * qml.Z(0), control = 1))
+        [AmplitudeEmbedding(array([1., 0.]), wires=[1]), Select(ops=(Z(0),), control=<Wires = [1]>), Adjoint(AmplitudeEmbedding(array([1., 0.]), wires=[1])), FlipSign((0,), wires=[0]), GlobalPhase(3.141592653589793, wires=[1])]
+
+        """
+
+        hamiltonian = kwargs["hamiltonian"]
+        control = kwargs["control"]
+
+        coeffs, unitaries = _positive_coeffs_hamiltonian(hamiltonian)
+
+        decomp_ops = []
+
+        decomp_ops.append(
+            qml.AmplitudeEmbedding(qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control)
+        )
+
+        decomp_ops.append(qml.Select(unitaries, control=control))
+        decomp_ops.append(
+            qml.adjoint(
+                qml.AmplitudeEmbedding(
+                    qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control
+                )
+            )
+        )
+
+        decomp_ops.append(qml.FlipSign(0, wires=control))
```

Comment:
It is kind of weird because Reflection is expecting an operator U but in this case is the identity. A possible equivalent
circuit could be this:

```python
        decomp_ops.append(qml.Reflection(qml.prod(*[qml.Identity(wire) for wire in control])))

```



Do you want me to update it?

Reply:
Identity can take multiple wires, so it should just be: 

```
decomp_ops.append(qml.Reflection(qml.Identity(control)))  # would we need to specify reflection wires ..? 
```

Reply:
Give it a try, I think it's better if we can use the tools we built and keep it modular. Unless you think there is a problem with performance.

Reply:
It worked, thanks!